crates/bevy_sprite/src/sprite.rs

```diff
@@ -1,19 +1,78 @@
 use crate::ColorMaterial;
 use bevy_asset::{Assets, Handle};
+use bevy_core::Bytes;
 use bevy_ecs::{Query, Res};
 use bevy_math::Vec2;
 use bevy_reflect::{Reflect, ReflectDeserialize, TypeUuid};
-use bevy_render::{renderer::RenderResources, texture::Texture};
+use bevy_render::{
+    renderer::{RenderResource, RenderResourceIterator, RenderResourceType, RenderResources},
+    texture::Texture,
+};
 use serde::{Deserialize, Serialize};
 
-#[derive(Debug, Default, Clone, RenderResources, TypeUuid, Reflect)]
+#[derive(Debug, Default, Clone, TypeUuid, Reflect)]
 #[uuid = "7233c597-ccfa-411f-bd59-9af349432ada"]
 pub struct Sprite {
     pub size: Vec2,
-    #[render_resources(ignore)]
+    pub flip_x: bool,
+    pub flip_y: bool,
     pub resize_mode: SpriteResizeMode,
 }
 
+impl RenderResources for Sprite {
```

Comment:
You can replace these RenderResources impls with:

```rust
#[derive(Debug, Default, Clone, TypeUuid, Reflect, RenderResources)]
#[render_resources(from_self)]
#[uuid = "7233c597-ccfa-411f-bd59-9af349432ada"]
pub struct Sprite {
    pub size: Vec2,
    pub flip_x: bool,
    pub flip_y: bool,
    pub resize_mode: SpriteResizeMode,
}
```

This does the same "pass through" logic to the RenderResource impl that the manual impls have.